src/router/validation/subscriptionValidator.test.ts

```diff
@@ -0,0 +1,208 @@
+            type: 'rest-hook',
+            endpoint,
+            payload: 'application/fhir+json',
+            header: ['Authorization: Bearer secret-token-abc-123'],
+        },
+    };
+};
+
+const getBundleResource = (resourceToAdd: { resourceType: string }) => {
+    return {
+        resourceType: 'Bundle',
+        type: 'transaction',
+        entry: [
+            {
+                fullUrl: 'urn:uuid:some-fake-uuid',
+                resource: invalidPatient,
+                request: {
+                    method: 'POST',
+                    url: 'Patient',
+                },
+            },
+            {
+                fullUrl: 'urn:uuid:another-fake-uuid',
+                resource: resourceToAdd,
+                request: {
+                    method: 'POST',
+                    url: resourceToAdd.resourceType,
+                },
+            },
+            {
+                request: {
+                    method: 'DELETE',
+                    url: 'Patient/some-fake-id',
+                },
+            },
+        ],
+    };
+};
+
+describe('Multi-tenancy mold: Validating Subscriptions', () => {
+    const allowedEndpoints: SubscriptionEndpoint[] = [
+        {
+            endpoint: 'https://fake-end-point-tenant1',
+            headers: ['header-name: header-value'],
+            tenantId: 'tenant1',
+        },
+        {
+            endpoint: new RegExp('^https://fake-end-point-tenant2'),
+            headers: ['header-name: header-value'],
+            tenantId: 'tenant2',
+        },
+    ];
+    const getAllowEndpoints: GetAllowEndpoints = jest.fn().mockResolvedValue(allowedEndpoints);
+    const subscriptionValidator = new SubscriptionValidator(search, getAllowEndpoints, true);
+
+    test('No error when validating valid Subscription resource', async () => {
+        const subscription = getSubscriptionResource('https://fake-end-point-tenant1');
+        await expect(subscriptionValidator.validate(subscription, 'tenant1')).resolves.toEqual(undefined);
+    });
+
+    test('No error when validating valid Bundle resource that contains valid Subscription resource', async () => {
+        const subscription = getSubscriptionResource('https://fake-end-point-tenant2');
+        const bundle = getBundleResource(subscription);
+        await expect(subscriptionValidator.validate(bundle, 'tenant2')).resolves.toEqual(undefined);
+    });
+
+    test('No error when validating resources that are not Subscription or Bundle', async () => {
+        await expect(subscriptionValidator.validate(invalidPatient, 'tenant1')).resolves.toEqual(undefined);
+    });
+
+    test('No error when validating Bundle that does not contain Subscription resource', async () => {
+        const bundle = getBundleResource(validPatient);
+        await expect(subscriptionValidator.validate(bundle, 'tenant2')).resolves.toEqual(undefined);
+    });
+
+    test('Show error when validating invalid Subscription resource', async () => {
```

Comment:
Can you add  couple more tests like this?
e.g. channel.type must be rest-hook, channel.payload must be application/fhir+json if present. etc.